db/migrate/20230602185438_create_publishing_pipeline_states.rb

```diff
@@ -0,0 +1,14 @@
+class CreatePublishingPipelineStates < ActiveRecord::Migration[7.0]
+  def change
+    create_table :publishing_pipeline_states do |t|
+      t.references :podcast, null: false, foreign_key: true
+      t.references :publishing_queue_item, null: false, foreign_key: true
+      t.integer :status, null: false, default: 0
+
+      t.datetime :created_at, null: false
+    end
+
+    add_index :publishing_pipeline_states, [:podcast_id, :publishing_queue_item_id, :status], unique: true, name: "index_publishing_pipeline_state_on_podcast_id_and_queue_item_id"
```

Comment:
Currently yes, that's correct.  For any given publishing attempt you cannot re-enter a state.  Like, you cannot start a pipeline twice, or have published the RSS twice in a single publishing attempt. Given the way the current code is structured, it's not something we do currently, right?   We'd fire off another Publishing job. And in this paradigm, we start up another PublishingPipeline (if we wanted to retry something).

There's nothing to prevent us from changing this, however, it's just a matter of articulating the case and adapting the constraints here.  Having a linearized set of state transitions (with the three end states) captures most of the current setup.

More radical change is possible too, if we wanted to segment these into completely separate jobs (Apple, OtherServiceX, OtherServiceY, RSS). But at that point we'd have to look handling those bigger state transitions between jobs at the level of the pipeline 'tick' (aka the say_when scheduler).